crates/runtime/src/dataconnector/databricks.rs

```diff
@@ -168,9 +170,114 @@ impl DataConnector for Databricks {
+                    dataconnector: "databricks".to_string(),
+                    source,
+                }))
+            }
+        };
+        let client = Arc::new(unity_catalog);
+
+        let mut dataset_params: SecretMap = catalog.dataset_params.clone().into();
+
+        let secrets_provider = runtime.secrets_provider();
+        let dataset_secret = match secrets_provider
+            .read()
+            .await
+            .get_secret("databricks")
+            .await
+            .map_err(|source| super::DataConnectorError::UnableToReadSecrets {
+                dataconnector: "databricks".to_string(),
+                source,
+            }) {
+            Ok(secret) => secret,
+            Err(e) => return Some(Err(e)),
+        };
+
+        if let Some(secret) = dataset_secret {
+            for (key, value) in secret.iter() {
+                dataset_params.insert(key.to_string(), value.clone());
+            }
+        }
+
+        let mode = self.params.get("mode").map(|v| v.expose_secret().as_str());
+        let (table_creator, table_reference_creator) = if let Some("delta_lake") = mode {
+            (
+                Arc::new(DeltaTableFactory::new(Arc::new(dataset_params.into_map())))
+                    as Arc<dyn Read>,
+                table_reference_creator_delta_lake as fn(UCTable) -> Option<TableReference>,
+            )
+        } else {
+            let normal_params: HashMap<String, String> = dataset_params
+                .iter()
+                .map(|(k, v)| (k.clone(), v.expose_secret().clone()))
+                .collect();
+            let dataset_databricks = match Databricks::new(None, Arc::new(normal_params))
+                .await
+                .map_err(
+                    |source| super::DataConnectorError::UnableToGetCatalogProvider {
+                        dataconnector: "databricks".to_string(),
+                        source: source.into(),
+                    },
+                ) {
+                Ok(dataset_databricks) => dataset_databricks,
+                Err(e) => return Some(Err(e)),
+            };
+
+            (
+                dataset_databricks.read_provider,
+                table_reference_creator_spark as fn(UCTable) -> Option<TableReference>,
+            )
+        };
+
+        let catalog_provider = match UnityCatalogProvider::try_new(
+            client,
+            CatalogId(catalog_id),
+            table_creator,
+            table_reference_creator,
+            catalog.include.clone(),
+        )
+        .await
+        {
+            Ok(provider) => provider,
+            Err(e) => {
+                return Some(Err(super::DataConnectorError::UnableToGetCatalogProvider {
+                    dataconnector: "databricks".to_string(),
+                    source: Box::new(e),
+                }))
+            }
+        };
+
+        Some(Ok(Arc::new(catalog_provider) as Arc<dyn CatalogProvider>))
     }
 }
+
+#[allow(clippy::unnecessary_wraps)]
```

Comment:
**nit:** This allow rule isn't allowing anything?

Reply:
It suppresses the rule that `Option<>` is unnecessary for the return value of this function. But we need it for the function pointer definition